### DISCUSSION TITLE
Constructor calling typescript vs java

### DISCUSSION THREAD

**Discussion by jonathangiber** [getgauge/gauge#2280] [General]
Hi guys,
Im using gauge in two different projects in the company I work.

One is in java and the new one in typescript.
While java would call and initialize the class of the step being called automatically, I noticed that in typescript this isn't happening and the constructor isn't being called at all.

Am I missing something or should I initialize the step class in every single step? 
Thanks

**Comment by zabil** [comment]
> initialize the class of the step being called automatically

Do you mean call the constructor?

**Reply by jonathangiber** [reply]
@zabil yes

**Reply by zabil** [reply]
Gauge doesn't rely on the constructor, so in this case it's most probably called only in the java implementation. So using a constructor may not work across language implementations.

**Reply by jonathangiber** [reply]
thanks!

**Comment by jonathangiber** [comment]
I wonder where I should initialize the scenarioStore, I thought creating a BaseStep in which every step class would inherit from it but guess it won't be convenient since no one is calling the step constructor classes.

Is the beforeSuite a good choice to init the scenarioStore then? thanks
